src/uwtools/scheduler.py

```diff
@@ -0,0 +1,128 @@
+"""
+Job Scheduling
+"""
+
+import logging
+import collections
+from typing import Dict
+
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s %(name)-12s %(levelname)-8s %(message)s",
+    handlers=[logging.StreamHandler()],
+)
+
+
+class JobCard(collections.UserList):
+    """represents a job card to submit to a scheduler"""
+
+    def content(self, line_separator: str = "\n") -> str:
+        """returns the formatted content of the job cards
+
+        Parameters
+        ----------
+        line_separator : str
+            the character or characters to join the content lines on.
+        """
+        return line_separator.join(self)
+
+
+class JobScheduler(collections.UserDict):
+    """object that creates JobCard"""
+
+    _map = {}
+    prefix = ""
+
+    def __getattr__(self, name):
+        if name in self:
+            return self[name]
```

Comment:
can you explain where is this used?

Reply:
This gives us the ability to use dot notation for any props.

object["job_name"] and object.job_name are both equivalent.

Reply:
Understood.
So if one were to use this `.` notation, we would have to implement the `getter` in every place we wish to use this.

Reply:
no, just here.  All the subclasses will inherit it.